### PR TITLE
[9.0][website_portal_invoice_separated] Separate sale and purchase portal documents.

### PR DESCRIPTION
**Pull request by yajo**
This module enables the sale and purchase portal, separates invoices between incoming and outgoing (with dedicated controllers), and hides the part that is not enabled depending on the contact company's type.

This will be WIP until these are merged:
- [x] https://github.com/OCA/server-tools/pull/442
- [x] #202
- [x] #204
- [x] #205
- [x] #206

@Tecnativa

### CODE REVIEW

File: website_portal_invoice_separated/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            # Get modified result
+            result = wrapped(*args, **kwargs)
+
+            # Restore request's env
+            request.context = old_context
+            del request.env
+
+            return result
+        return wrapper
+    return superwrapper
```
Comment (yajo) [inline website_portal_invoice_separated/controllers/main.py:37]: @kutyel, you will like to see this :wink:

Comment (kutyel) [inline website_portal_invoice_separated/controllers/main.py:37]: Nice! thanks for mentioning :smile:

File: website_portal_invoice_separated/views/layout.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- © 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+     License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). -->
+<odoo>
+    <!-- Show purchase documents only for supplier users -->
+    <template id="portal_layout_purchase"
+              name="Show only for suppliers"
+              inherit_id="website_portal_purchase.portal_layout">
+        <xpath expr="//t[@name='supplier_menu']" position="attributes">
+            <attribute name="t-if">user.commercial_partner_id.supplier</attribute>
```
Comment (yajo) [inline website_portal_invoice_separated/views/layout.xml:10]: @carlosdauden @sergio-teruel This is what we use now to check permissions.

**Comment by rafaelbn** [comment]
@Yajo @sergio-teruel @carlosdauden https://github.com/OCA/website/pull/202 is merged. Just https://github.com/OCA/website/pull/206 for this one. Please take in care. Thanks

**Comment by pedrobaeza** [comment]
@chienandalu @luismontalba This one is ready for being reviewed

**Comment by chienandalu** [comment]
runbot can't build the image

**Comment by yajo** [comment]
Yes, @moylop260 could you review what's happening with website 9.0 runbot please?

**Comment by moylop260** [comment]
@Yajo 
I have created the following issue: https://github.com/OCA/runbot-addons/issues/129

**Comment by moylop260** [comment]
@Yajo 
FYI now runbot is green.
The issue was fixed!
Thanks for reporting

**Comment by pedrobaeza** [comment]
Closing this due to no continuation in the project. If somebody has interest on it, he/she can open another PR taking this work.

**Pull request closed without merging**